hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -335,7 +337,18 @@ public interface HoodieTimeline extends Serializable {
   /**
    * @return Get the stream of completed instants
    */
-  Stream<HoodieInstant> getInstants();
+  default Stream<HoodieInstant> getInstantsAsStream() {
+    return getInstants().stream();
+  }
```

Comment:
can we follow the existing convention to keep default impl. in HoodieDefaultTimeline? `getInstants()` is not implemented at this level hence `getInstantsAsStream()` may not be optimized to directly call `stream()`; for e.g. there could be a timeline subclass which is backed by stream of instants so you can return the stream directly instead of `getInstants().stream()`. let's leave the impl. to subclasses